code/modules/reagents/dispenser/dispenser_presets.dm

```diff
@@ -113,3 +113,22 @@
 			/obj/item/weapon/reagent_containers/chem_disp_cartridge/ale,
 			/obj/item/weapon/reagent_containers/chem_disp_cartridge/mead
 		)
+
+/obj/machinery/chemical_dispenser/bar_coffee
+	name = "coffee dispenser"
+	desc = "Driving crack dealers out of employment since 2280."
+	icon_state = "coffee_dispenser"
+	ui_title = "Coffee Dispenser"
+	accept_drinking = 1
+
+/obj/machinery/chemical_dispenser/bar_coffee/full
+	spawn_cartridges = list(
+			/obj/item/weapon/reagent_containers/chem_disp_cartridge/coffee,
+			/obj/item/weapon/reagent_containers/chem_disp_cartridge/cafe_latte,
+			/obj/item/weapon/reagent_containers/chem_disp_cartridge/soy_latte,
+			/obj/item/weapon/reagent_containers/chem_disp_cartridge/hot_coco,
+			/obj/item/weapon/reagent_containers/chem_disp_cartridge/milk,
+			/obj/item/weapon/reagent_containers/chem_disp_cartridge/cream,
+			/obj/item/weapon/reagent_containers/chem_disp_cartridge/tea,
+			/obj/item/weapon/reagent_containers/chem_disp_cartridge/ice

```

Comment:
Cartridges will not regenerate over time with this setup; you'll need to add the type in dispenser2_energy.dm for that to happen.